ecommerce_api_client/tests/test_auth.py

```diff
@@ -18,32 +24,57 @@ def setUp(self):
         self.signing_key = 'edx'
         httpretty.register_uri(httpretty.GET, self.url)
 
-    def assert_expected_token_value(self, tracking_context=None):
+    def assert_expected_token_value(self, tracking_context=None, issuer=None, expires_in=30):
         """ DRY helper. """
 
         # Mock the HTTP response and issue the request
-        auth_kwargs = {'tracking_context': tracking_context} if tracking_context else {}
-        requests.get(self.url, auth=JwtAuth(self.username, self.full_name, self.email, self.signing_key, **auth_kwargs))
+        auth_kwargs = {'expires_in': expires_in}
 
-        # Verify the header was set as expected on the request
         data = {
             'username': self.username,
             'full_name': self.full_name,
             'email': self.email
         }
-        data.update(auth_kwargs)
+
+        if issuer:
+            auth_kwargs['issuer'] = issuer
+            data['iss'] = issuer
+
+        if tracking_context:
+            auth_kwargs['tracking_context'] = tracking_context
+            data['tracking_context'] = tracking_context
+
+        if expires_in:
+            data['exp'] = CURRENT_TIME + datetime.timedelta(seconds=expires_in)
+
+        requests.get(self.url, auth=JwtAuth(self.username, self.full_name, self.email, self.signing_key, **auth_kwargs))
+
+        # Verify the header was set as expected on the request
         token = jwt.encode(data, self.signing_key)
         self.assertEqual(httpretty.last_request().headers['Authorization'], 'JWT {}'.format(token))
 
     @httpretty.activate
+    @patch.object(JwtAuth, 'datetime_now', Mock(return_value=CURRENT_TIME))
     def test_headers(self):
         """ Verify the class adds an Authorization header that includes the correct JWT. """
         self.assert_expected_token_value()
 
     @httpretty.activate
+    @patch.object(JwtAuth, 'datetime_now', Mock(return_value=CURRENT_TIME))
```

Comment:
You should not need to mock `datetime_now` to validate the tracking_context scope.